src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs

```diff
@@ -534,13 +534,18 @@ private HttpClient CreateHttpClient()
                     httpClientHandler.Proxy = _httpConnectionOptions.Proxy;
                 }
 
-                // Only access HttpClientHandler.ClientCertificates and HttpClientHandler.CookieContainer
-                // if the user has configured those options
-                // Some variants of Mono do not support client certs or cookies and will throw NotImplementedException
-                if (_httpConnectionOptions.Cookies.Count > 0)
+                try
                 {
+                    // On supported platforms, we need to pass the cookie container to the http client
+                    // so that we can capture any cookies from the negotiate response and give them to WebSockets.
                     httpClientHandler.CookieContainer = _httpConnectionOptions.Cookies;
                 }
+                // Some variants of Mono do not support client certs or cookies and will throw NotImplementedException or NotSupportedException
+                // Also WASM doesn't support some settings in the browser
+                catch (Exception ex) when (ex is NotSupportedException || ex is NotImplementedException) { }
```

Comment:
Can we log something to the effect that cookies are unsupported on this platform?

Reply:
It'd have to be debug or trace, because it'll always log this on some platforms even if cookies aren't being used.

Reply:
That's fine with me.

Reply:
It'd still be logged at most once per connection, right?

Reply:
Oh... it'll be logged once per `StartAsyncCore` call which is called from `StartAsync` and reconnects.

Reply:
If it's easy, it would be nice to only log this on during the initial start and not reconnect attempts.

Reply:
Added a static for logging, ugly but works 🤷‍♀️

Reply:
Can we leave it at once per connection? I don't want to be in a situation where we're investigating logs and we don't see "Cookies are not supported on this platform" so we don't think that's a problem, but it was really already logged by a previous connection.

Reply:
That's not possible with the layering we have

Reply:
If not once per start, what about once per HubConnection?

Reply:
Same issue. HubConnection creates a new connection for starts and reconnects.

Reply:
I see. Then I prefer to log it every time, but trace level is fine.